In [92]:
import pandas as pd
#import file
reactions = pd.read_csv('/Users/candicewu/Desktop/DA Projects/6./Reactions.csv')
reactions.head()


Unnamed: 0                            Content ID  \
0           0  97522e57-d9ab-4bd6-97bf-c24d952602d2   
1           1  97522e57-d9ab-4bd6-97bf-c24d952602d2   
2           2  97522e57-d9ab-4bd6-97bf-c24d952602d2   
3           3  97522e57-d9ab-4bd6-97bf-c24d952602d2   
4           4  97522e57-d9ab-4bd6-97bf-c24d952602d2   

                                User ID     Type             Datetime  
0                                   NaN      NaN  2021-04-22 15:17:15  
1  5d454588-283d-459d-915d-c48a2cb4c27f  disgust  2020-11-07 09:43:50  
2  92b87fa5-f271-43e0-af66-84fac21052e6  dislike  2021-06-17 12:22:51  
3  163daa38-8b77-48c9-9af6-37a6c1447ac2   scared  2021-04-18 05:13:58  
4  34e8add9-0206-47fd-a501-037b994650a2  disgust  2021-01-06 19:13:01

In [93]:
#delete the first column by using ilo[]
reactions=reactions.iloc[:,1:]
reactions.head()

Content ID                               User ID  \
0  97522e57-d9ab-4bd6-97bf-c24d952602d2                                   NaN   
1  97522e57-d9ab-4bd6-97bf-c24d952602d2  5d454588-283d-459d-915d-c48a2cb4c27f   
2  97522e57-d9ab-4bd6-97bf-c24d952602d2  92b87fa5-f271-43e0-af66-84fac21052e6   
3  97522e57-d9ab-4bd6-97bf-c24d952602d2  163daa38-8b77-48c9-9af6-37a6c1447ac2   
4  97522e57-d9ab-4bd6-97bf-c24d952602d2  34e8add9-0206-47fd-a501-037b994650a2   

      Type             Datetime  
0      NaN  2021-04-22 15:17:15  
1  disgust  2020-11-07 09:43:50  
2  dislike  2021-06-17 12:22:51  
3   scared  2021-04-18 05:13:58  
4  disgust  2021-01-06 19:13:01

In [94]:
#inspect data type for each column and the number of missing values
reactions.info(verbose=True, show_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25553 entries, 0 to 25552
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Content ID  25553 non-null  object
 1   User ID     22534 non-null  object
 2   Type        24573 non-null  object
 3   Datetime    25553 non-null  object
dtypes: object(4)
memory usage: 798.7+ KB


According to the output, the total number of rows should be 25553. However, there are only 24573 rows for 'Type' column, which suggests that there may be some missing values in this column

In [95]:
#remove missing values by using the dropna() function
reactions=reactions.dropna(subset=['Type'])
reactions.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 24573 entries, 1 to 25552
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Content ID  24573 non-null  object
 1   User ID     22534 non-null  object
 2   Type        24573 non-null  object
 3   Datetime    24573 non-null  object
dtypes: object(4)
memory usage: 959.9+ KB


In [96]:
#delect irrelevant columns by using the drop() fcunction
reactions=reactions.drop(columns=['User ID'])
reactions.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 24573 entries, 1 to 25552
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Content ID  24573 non-null  object
 1   Type        24573 non-null  object
 2   Datetime    24573 non-null  object
dtypes: object(3)
memory usage: 767.9+ KB


In [97]:
#rename column name by using the rename() function
reactions=reactions.rename(columns={"Type": "Reaction Type"})

In [98]:
#import file
content = pd.read_csv('/Users/candicewu/Desktop/DA Projects/6./Content.csv')
content.head()

Unnamed: 0                            Content ID  \
0           0  97522e57-d9ab-4bd6-97bf-c24d952602d2   
1           1  9f737e0a-3cdd-4d29-9d24-753f4e3be810   
2           2  230c4e4d-70c3-461d-b42c-ec09396efb3f   
3           3  356fff80-da4d-4785-9f43-bc1261031dc6   
4           4  01ab84dd-6364-4236-abbb-3f237db77180   

                                User ID   Type        Category  \
0  8d3cd87d-8a31-4935-9a4f-b319bfe05f31  photo        Studying   
1  beb1f34e-7870-46d6-9fc7-2e12eb83ce43  photo  healthy eating   
2  a5c65404-5894-4b87-82f2-d787cbee86b4  photo  healthy eating   
3  9fb4ce88-fac1-406c-8544-1a899cee7aaf  photo      technology   
4  e206e31b-5f85-4964-b6ea-d7ee5324def1  video            food   

                                                 URL  
0  https://socialbuzz.cdn.com/content/storage/975...  
1  https://socialbuzz.cdn.com/content/storage/9f7...  
2  https://socialbuzz.cdn.com/content/storage/230...  
3  https://socialbuzz.cdn.com/content/storage/356...  
4  https://socialbuzz.cdn.com/content/storage/01a...

In [99]:
#delete the first column by using ilo[]
content=content.iloc[:,1:]
content.head()

Content ID                               User ID  \
0  97522e57-d9ab-4bd6-97bf-c24d952602d2  8d3cd87d-8a31-4935-9a4f-b319bfe05f31   
1  9f737e0a-3cdd-4d29-9d24-753f4e3be810  beb1f34e-7870-46d6-9fc7-2e12eb83ce43   
2  230c4e4d-70c3-461d-b42c-ec09396efb3f  a5c65404-5894-4b87-82f2-d787cbee86b4   
3  356fff80-da4d-4785-9f43-bc1261031dc6  9fb4ce88-fac1-406c-8544-1a899cee7aaf   
4  01ab84dd-6364-4236-abbb-3f237db77180  e206e31b-5f85-4964-b6ea-d7ee5324def1   

    Type        Category                                                URL  
0  photo        Studying  https://socialbuzz.cdn.com/content/storage/975...  
1  photo  healthy eating  https://socialbuzz.cdn.com/content/storage/9f7...  
2  photo  healthy eating  https://socialbuzz.cdn.com/content/storage/230...  
3  photo      technology  https://socialbuzz.cdn.com/content/storage/356...  
4  video            food  https://socialbuzz.cdn.com/content/storage/01a...

In [100]:
content.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Content ID  1000 non-null   object
 1   User ID     1000 non-null   object
 2   Type        1000 non-null   object
 3   Category    1000 non-null   object
 4   URL         801 non-null    object
dtypes: object(5)
memory usage: 39.2+ KB


In [101]:
#remove irrelevant columns by using the drop() function
content=content.drop(columns=['URL', 'User ID'])
content.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Content ID  1000 non-null   object
 1   Type        1000 non-null   object
 2   Category    1000 non-null   object
dtypes: object(3)
memory usage: 23.6+ KB


In [102]:
#replaces a specified phrase with another specified phrase by using the replace() function
content['Category']=(content['Category'].str.replace(" "" ", "",regex=True))
content

Content ID   Type         Category
0    97522e57-d9ab-4bd6-97bf-c24d952602d2  photo         Studying
1    9f737e0a-3cdd-4d29-9d24-753f4e3be810  photo   healthy eating
2    230c4e4d-70c3-461d-b42c-ec09396efb3f  photo   healthy eating
3    356fff80-da4d-4785-9f43-bc1261031dc6  photo       technology
4    01ab84dd-6364-4236-abbb-3f237db77180  video             food
..                                    ...    ...              ...
995  b4cef9ef-627b-41d7-a051-5961b0204ebb  video  public speaking
996  7a79f4e4-3b7d-44dc-bdef-bc990740252c    GIF       technology
997  435007a5-6261-4d8b-b0a4-55fdc189754b  audio         veganism
998  4e4c9690-c013-4ee7-9e66-943d8cbd27b7    GIF          culture
999  75d6b589-7fae-4a6d-b0d0-752845150e56  audio       technology

[1000 rows x 3 columns]

In [103]:
#rename column names by using the rename( ) function
content= content.rename(columns={"Type": "Content Type"})
content.head()

Content ID Content Type        Category
0  97522e57-d9ab-4bd6-97bf-c24d952602d2        photo        Studying
1  9f737e0a-3cdd-4d29-9d24-753f4e3be810        photo  healthy eating
2  230c4e4d-70c3-461d-b42c-ec09396efb3f        photo  healthy eating
3  356fff80-da4d-4785-9f43-bc1261031dc6        photo      technology
4  01ab84dd-6364-4236-abbb-3f237db77180        video            food

In [104]:
#import file
reactiontypes = pd.read_csv('/Users/candicewu/Desktop/DA Projects/6./ReactionTypes.csv')
reactiontypes.head()

Unnamed: 0        Type Sentiment  Score
0           0       heart  positive     60
1           1        want  positive     70
2           2     disgust  negative      0
3           3        hate  negative      5
4           4  interested  positive     30

In [105]:
#delection the first column by using the iloc() function
reactiontypes=reactiontypes.iloc[:,1:]
reactiontypes.head()

Type Sentiment  Score
0       heart  positive     60
1        want  positive     70
2     disgust  negative      0
3        hate  negative      5
4  interested  positive     30

In [106]:
#check
reactiontypes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16 entries, 0 to 15
Data columns (total 3 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Type       16 non-null     object
 1   Sentiment  16 non-null     object
 2   Score      16 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 512.0+ bytes


In [107]:
#to merge two dataset by using the merge () function
df = pd.merge(
    reactions,
    reactiontypes,
    left_on=["Reaction Type"],
    right_on=["Type"],
    how="left",
)
df

Content ID Reaction Type  \
0      97522e57-d9ab-4bd6-97bf-c24d952602d2       disgust   
1      97522e57-d9ab-4bd6-97bf-c24d952602d2       dislike   
2      97522e57-d9ab-4bd6-97bf-c24d952602d2        scared   
3      97522e57-d9ab-4bd6-97bf-c24d952602d2       disgust   
4      97522e57-d9ab-4bd6-97bf-c24d952602d2    interested   
...                                     ...           ...   
24568  75d6b589-7fae-4a6d-b0d0-752845150e56       dislike   
24569  75d6b589-7fae-4a6d-b0d0-752845150e56     intrigued   
24570  75d6b589-7fae-4a6d-b0d0-752845150e56    interested   
24571  75d6b589-7fae-4a6d-b0d0-752845150e56       worried   
24572  75d6b589-7fae-4a6d-b0d0-752845150e56       cherish   

                  Datetime        Type Sentiment  Score  
0      2020-11-07 09:43:50     disgust  negative      0  
1      2021-06-17 12:22:51     dislike  negative     10  
2      2021-04-18 05:13:58      scared  negative     15  
3      2021-01-06 19:13:01     disgust  negative      0  
4      2020-08-23 12:25:58  interested  positive     30  
...                    ...         ...       ...    ...  
24568  2020-06-27 09:46:48     dislike  negative     10  
24569  2021-02-16 17:17:02   intrigued  positive     45  
24570  2020-09-12 03:54:58  interested  positive     30  
24571  2020-11-04 20:08:31     worried  negative     12  
24572  2021-01-04 04:55:11     cherish  positive     70  

[24573 rows x 6 columns]

In [108]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 24573 entries, 0 to 24572
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Content ID     24573 non-null  object
 1   Reaction Type  24573 non-null  object
 2   Datetime       24573 non-null  object
 3   Type           24573 non-null  object
 4   Sentiment      24573 non-null  object
 5   Score          24573 non-null  int64 
dtypes: int64(1), object(5)
memory usage: 1.3+ MB


In [109]:
final_df = pd.merge(
   df,
   content,
    on=["Content ID"],
    how="left",
)
final_df

Content ID Reaction Type  \
0      97522e57-d9ab-4bd6-97bf-c24d952602d2       disgust   
1      97522e57-d9ab-4bd6-97bf-c24d952602d2       dislike   
2      97522e57-d9ab-4bd6-97bf-c24d952602d2        scared   
3      97522e57-d9ab-4bd6-97bf-c24d952602d2       disgust   
4      97522e57-d9ab-4bd6-97bf-c24d952602d2    interested   
...                                     ...           ...   
24568  75d6b589-7fae-4a6d-b0d0-752845150e56       dislike   
24569  75d6b589-7fae-4a6d-b0d0-752845150e56     intrigued   
24570  75d6b589-7fae-4a6d-b0d0-752845150e56    interested   
24571  75d6b589-7fae-4a6d-b0d0-752845150e56       worried   
24572  75d6b589-7fae-4a6d-b0d0-752845150e56       cherish   

                  Datetime        Type Sentiment  Score Content Type  \
0      2020-11-07 09:43:50     disgust  negative      0        photo   
1      2021-06-17 12:22:51     dislike  negative     10        photo   
2      2021-04-18 05:13:58      scared  negative     15        photo   
3      2021-01-06 19:13:01     disgust  negative      0        photo   
4      2020-08-23 12:25:58  interested  positive     30        photo   
...                    ...         ...       ...    ...          ...   
24568  2020-06-27 09:46:48     dislike  negative     10        audio   
24569  2021-02-16 17:17:02   intrigued  positive     45        audio   
24570  2020-09-12 03:54:58  interested  positive     30        audio   
24571  2020-11-04 20:08:31     worried  negative     12        audio   
24572  2021-01-04 04:55:11     cherish  positive     70        audio   

         Category  
0        Studying  
1        Studying  
2        Studying  
3        Studying  
4        Studying  
...           ...  
24568  technology  
24569  technology  
24570  technology  
24571  technology  
24572  technology  

[24573 rows x 8 columns]

In [110]:
final_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 24573 entries, 0 to 24572
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Content ID     24573 non-null  object
 1   Reaction Type  24573 non-null  object
 2   Datetime       24573 non-null  object
 3   Type           24573 non-null  object
 4   Sentiment      24573 non-null  object
 5   Score          24573 non-null  int64 
 6   Content Type   24573 non-null  object
 7   Category       24573 non-null  object
dtypes: int64(1), object(7)
memory usage: 1.7+ MB


In [111]:
final_df.to_csv("/Users/candicewu/Desktop/cleaned.csv", index=False)